### PR TITLE
[Curl] Fix cmake target name

### PR DESCRIPTION
**Pull request by PhoebeHui**
My fault, I updated CURLTargets-release.cmake in #6429, but suddenly removed by following commits, Sorry for that. Related issue https://github.com/microsoft/vcpkg/issues/6643

### CODE REVIEW

File: ports/curl/portfile.cmake [hunk 1]
```diff
@@ -134,9 +134,9 @@ if(EXISTS "${CURRENT_PACKAGES_DIR}/bin/curl${EXECUTABLE_SUFFIX}")
     vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/curl)
 
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
-        file(READ "${CURRENT_PACKAGES_DIR}/share/curl/curl-target-release.cmake" RELEASE_MODULE)
+        file(READ "${CURRENT_PACKAGES_DIR}/share/curl/CURLTargets-release.cmake" RELEASE_MODULE)
```
Comment (Neumann-A) [inline ports/curl/portfile.cmake]: ähm why are you changing that? These are automatic generated filenames. Please do not change them! 

```
if(NOT CURL_FIND_COMPONENTS)
  set(CURL_FIND_COMPONENTS curl libcurl)
```
```
foreach(_comp ${CURL_FIND_COMPONENTS})
  if(EXISTS "${_DIR}/${_comp}-target.cmake")
    include("${_DIR}/${_comp}-target.cmake")
```
```
file(GLOB CONFIG_FILES "${_DIR}/libcurl-target-*.cmake")
foreach(f ${CONFIG_FILES})
  include(${f})
endforeach()

```

Comment (PhoebeHui) [reply ports/curl/portfile.cmake]: Sorry for the confuse, recently we updated curl to 7.65.0, the cmake target name updated in curl, related commit is https://github.com/curl/curl/commit/69328490fc91bfa00a6a90da84239524b01f5527, so we need  update it as well.

Comment (Neumann-A) [reply ports/curl/portfile.cmake]: Ah okay the target files have been renamed to uppercase. Was this an upstream change?

Comment (Neumann-A) [reply ports/curl/portfile.cmake]: What happens if you directly set CMAKE_INSTALL_BINDIR to the correct folder in the configure step? Then the rewrite in the config is probably not needed. The mentioned upstream commit mentions GNUInstallDirs so it is probably worth a try

Comment (PhoebeHui) [reply ports/curl/portfile.cmake]: Yes, you can see the changes In curl CMakeLists.txt line #1158 to line #1162.

**Pull request merged**